payments/views.py

```diff
@@ -15,18 +15,49 @@ class WorkshopList(ListView):
     context_object_name = 'upcoming_workshops'
 
 
-class WorkshopDetail(DetailView):
+class WorkshopDetail(FormMixin, DetailView):
     model = Workshop
+    form_class = OrderForm
     context_object_name = 'workshop'
     template_name = 'payments/detail.html'
 
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['workshop'] = self.object
```

Comment:
Gotcha - the kwargs get passed into the form, and our form (`OrderForm`) is expecting the `workshops` kwarg. The flow for the `FormMixin`: in line 31 we call `self.get_form()` (provided by the mixin), which in turn calls `self.get_form_kwargs()` (also provided by the mixin, but we are overriding), and the values returned from that get passed into `OrderForm`. Then we use the `workshop` to populate the dynamic form fields.